### ISSUE TITLE
Add option to specify Queue ID

### ISSUE THREAD

**Issue by user273835** [huserben/TfsExtensions#30]
It would be nice to be able to specify the Queue ID in which the triggered build will run.  Use case:  I have a build I want to run in multiple queues to verify agent performance and requirements, where each queue has agents running different base operating systems.

https://www.visualstudio.com/en-us/docs/integrate/api/build/builds#queue-a-build
![image](https://user-images.githubusercontent.com/17420310/31080420-42d1b4bc-a74e-11e7-8fce-bc4ed57fefc6.png)

**Comment by huserben** [comment]
Hi

thanks for the input.
I will include the functionality in the next version.

I'll let you know once I uploaded it.

**Comment by user273835** [comment]
Thanks!

**Comment by huserben** [comment]
Hi

I just updated the task, you should now find the Queue ID under the advanced configuration section.
If not specified, it will use the default, otherwise it'll try to use the one specified. If the queue with the specified id does not exist, it will give an approrpiate error message.

Please try it out and let me know if everything works as you would expect. Otherwise feel free to let me know that it's working or just close the issue :-)

Thanks again for the input 👍

**Comment by user273835** [comment]
This is great and works as expected.  My only other enhancement request would be to make the Queue ID field a dropdown or to allow the name of the queue to be entered as a string (so when I look at the build later, I don't wonder which queue 153 is).  If you don't want to add that, I understand.  I'll wait for your feedback before closing the issue.

**Comment by user273835** [comment]
Making it a name would be very nice.  I don't have any timetable.  Do you want to track that in a new issue and close this one?

**Comment by huserben** [comment]
Hi

yeah I will create a new issur for it. Once this is done i then close this one.

Thanks again for your feedback

**Comment by user273835** [comment]
You're welcome.  I appreciate the quick response.

**Comment by huserben** [comment]
I added the new issue (https://github.com/huserben/TfsExtensions/issues/33) and will close this one then.
